docs/reference/configuration/configuration.md

```diff
@@ -1,48 +1,276 @@
+- `target_overrides` is a dictionary with target-specific values for the configuration parameters.
 
-2. Construct the thread like this in your application:
+All configuration parameters defined in `mylib` have a `mylib.` prefix. In `mbed_app.json`, `buffer_size` is accessible using the name `mylib.buffer_size`.
 
-    `Thread thread(osPriorityNormal, 2000);`
+Use `target_overrides` to override the values of the parameters, depending on the current compilation target. The configuration system matches keys in `target_overrides` against target labels. (You can find a description of Mbed targets in our documentation about [adding and configuring targets](/docs/development/tools/adding-and-configuring-targets.html).) If a key inside `target_overrides` matches one of the target labels, the parameter values change according to the value of the key. 
 
-By default, mbed OS dynamically allocates the memory for the thread's stack from the global heap, though you can change it in the [Thread constructor](/docs/development/reference/thread.html). Here is an example application that uses memory allocated from the main thread's stack for the newly spawned thread stack:
+It is an error for `mbed_lib.json` to override configuration parameters that were not defined.
 
+#### Overriding target attributes
+
+Target configurations contain a set of attributes that you may manipulate with configuration. You may override these attributes as if they were a normal configuration parameter. Attributes may be cumulative, in which case they are a list of items. You may add to a cumulative attribute by overriding a configuration parameter with the name of the cumulative attribute suffixed with `_add` and remove from a cumulative attribute with the suffix `_remove`. When you override, add to or subtract from a cumulative attribute, the value must be a list of items to replace the definition with, add or remove. For example, add the value `IPV4` to a target's features list with the syntax:
+
+```JSON
+"target.features_add": ["IPV4"]
 ```
-int main() {
-    uint32_t stack_on_main_stack[128 / sizeof(uint32_t)]; // 128 bytes
-    Thread thread(osPriorityNormal, sizeof(stack_on_main_stack), stack_on_main_stack);
+
+It is an error to both add and subtract the same value from a cumulative attribute. For a list of the attributes that you may overwrite, please see our documentation about [adding and configuring targets](/docs/development/tools/adding-and-configuring-targets.html)
+
+### `mbed_app.json` Specification
+
+`mbed_app.json` may be present at the root of your application or specified as the argument to the `--app-config` parameter. The configuration system interprets only one `mbed_app.json` during `mbed compile` or `mbed export`, unlike library configuration. Like `mbed_lib.json`, `mbed_app.json` is a JSON formatted document that contains a root JSON Object. The keys within this object are sections. The allowed sections and their meanings are below:
```

Comment:
I think it'd be worth mentioning `mbed compile` and `mbed export` with regards to the `--app-config` parameter.

Might take a little rewording though, something like: 

> `mbed_app.json` may be present at the root of your application or specified as the argument to the `--app-config` parameter when using `mbed compile` and `mbed export`. The configuration system interprets only one `mbed_app.json`, unlike library configuration.

Reply:
Also, same deal about "sections" here.